lib/tasks/admin.rake

```diff
@@ -181,4 +182,72 @@ namespace :admin do
   task purge_job: :environment do
     PurgeJob.perform_later
   end
+
+  desc 'Add API OAuth Application for Backend Services API Workflow'
+  task create_oauth_app_for_backend_services_workflow: :environment do
+    # Read from JSON file with needed information
+    begin
+      file = File.read(ENV["API_FILE_PATH"])
+      data = JSON.parse(file)
+    rescue => error
+      next puts "Error reading from expected JSON file that contains needed data: #{error}"
+    end
+
+    # Validation of needed data
+    next puts "Error! File does not contain required 'app_name' field." if data["app_name"].nil?
+    next puts "Error! File does not contain required 'email' field." if data["email"].nil?
+    next puts "Error! File does not contain required 'jurisdiction_path' field." if data["jurisdiction_path"].nil?
+    next puts "Error! File does not contain required 'public_key_set' field." if data["public_key_set"].nil?
+    next puts "Error! File does not contain required 'scopes' field." if data["scopes"].nil?
+
+    APP_NAME = data["app_name"]
+    EMAIL = data["email"]
+    JURISDICTION_PATH = data["jurisdiction_path"]
+    PUBLIC_KEY_SET = data["public_key_set"]
+    SCOPES = data["scopes"]
+
+    # Optional value for this workflow - should only include if client wants to use user workflow as well.
+    REDIRECT_URI = data["redirect_uri"] || 'urn:ietf:wg:oauth:2.0:oob'
+
+    jurisdiction = Jurisdiction.find_by(path: JURISDICTION_PATH)
+    next puts "Error! JURISDICTION_PATH is invalid: #{JURISDICTION_PATH}" unless jurisdiction.present?
+
+    user_already_exists = User.find_by(email: EMAIL).present?
+    next puts "Error! User with email #{EMAIL} already exists in the system." if user_already_exists
+
+    # Create shadow user for application
+    begin
+      app_user = User.create!(
+        email: EMAIL,
+        password: User.rand_gen,
```

Comment:
Would things still function if we locked the shadow account from the start?

Reply:
I THINK so - locked accounts just prevent them from logging in as far as I understand it, not sure it would cause any other issues.

Reply:
Okay, I'm thinking it might be worth doing a `app_user.lock_access!` before the puts and after the save

Reply:
One repercussions is that these "users" wouldn't receive release emails which I suppose the emails they're tied to might want to receive? Just thinking through. Either way I can test and make sure there's no other repercussions of doing so. We don't send them their temp password so they should never be able to log in anyway - but would be a nice extra precaution

Reply:
If no one is actually going to log in with it, it should definitely be locked

Reply:
Done.

Reply:
thanks!